lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,277 +5,204 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 import React from 'react';
-import { mount } from 'enzyme';
-import { faChevronCircleDown, faChevronCircleUp } from '@fortawesome/free-solid-svg-icons';
 import { act } from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
 
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import NxAccordion, { Props } from '../NxAccordion';
-import NxButton from '../../NxButton/NxButton';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
 
-function createClickEvent() {
-  return new MouseEvent('click', { bubbles: true });
-}
+import NxAccordion from '../NxAccordion';
+import NxButton from '../../NxButton/NxButton';
+import { render } from '@testing-library/react';
 
 describe('NxAccordion', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxAccordion, {}),
-      getMountedComponent = enzymeUtils.getMountedComponent<Props>(NxAccordion, {});
+  const quickRender = rtlRender(NxAccordion, {});
+  const renderEl = rtlRenderElement(NxAccordion, {});
 
   it('renders a <details> element with the provided props', function() {
-    const component = getShallowComponent({ id: 'foo', open: true, title: 'test title' });
-
-    expect(component).toMatchSelector('details#foo');
-    expect(component).toHaveProp('open', true);
-    expect(component).toHaveProp('title', 'test title');
+    expect(quickRender({ id: 'foo' }).getByRole('group')).toHaveAttribute('id', 'foo');
+    expect(renderEl({ open: true })).toHaveAttribute('open');
+    expect(renderEl({ title: 'test title' })).toHaveAttribute('title', 'test title');
   });
 
-  it('sets the nx-accordion class along with any provided className', function() {
-    expect(getShallowComponent()).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('nx-accordion');
-    expect(getShallowComponent({ className: 'foo' })).toHaveClassName('foo');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
```

Comment:
@rpokorny Didn't know how to link this in your other comment, but is this what you meant by _'I'd replace this with our usual style of test for custom classnames that checks that all classes present without a custom classname are also still present with a custom classname.'_?